diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06f5395..e815a3e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,8 +120,15 @@ if(MSVC)
   option(SHADERC_ENABLE_SHARED_CRT
           "Use the shared CRT instead of the static CRT"
 		  OFF)
+  option(SHADERC_ENABLE_SHARED_DEBUG_CRT
+		  "Use the shared debug CRT instead of the static CRT"
+		  OFF)
   if (SHADERC_ENABLE_SHARED_CRT)
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+    if (SHADERC_ENABLE_SHARED_DEBUG_CRT)
+      set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDebugDLL")
+	else()
+	  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+	endif()
   else()
     set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
   endif()
diff --git a/libshaderc/CMakeLists.txt b/libshaderc/CMakeLists.txt
index df9a88d..572822c 100644
--- a/libshaderc/CMakeLists.txt
+++ b/libshaderc/CMakeLists.txt
@@ -43,7 +43,7 @@ target_compile_definitions(shaderc_shared
 )
 set_target_properties(shaderc_shared PROPERTIES SOVERSION 1)
 
-if(SHADERC_ENABLE_INSTALL)
+if(SHADERC_ENABLE_INSTALL AND NOT SHADERC_ONLY_COMBINED)
   install(
     FILES
       include/shaderc/env.h
@@ -59,7 +59,7 @@ if(SHADERC_ENABLE_INSTALL)
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     BUNDLE DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-endif(SHADERC_ENABLE_INSTALL)
+endif(SHADERC_ENABLE_INSTALL AND NOT SHADERC_ONLY_COMBINED)
 
 find_package(Threads)
 set(SHADERC_LIBS
